dataservice/api/demographic/resources.py

```diff
@@ -0,0 +1,150 @@
+import os
+
+from flask import request
+from flask_restplus import Namespace, Resource
+from sqlalchemy.exc import IntegrityError
+
+from dataservice.extensions import db
+from dataservice.api.demographic import models
+from dataservice.api.common.formatters import kf_response
+from dataservice.api.common.error_handler import (
+    handle_invalid_input,
+    handle_not_found
+)
+
+THIS_DIR = os.path.abspath(os.path.dirname(__file__))
+README_FILE = os.path.join(THIS_DIR, 'README.md')
+
+description = open(README_FILE).read()
+demographic_api = Namespace(name='demographics', description=description)
+
+from dataservice.api.demographic.serializers import (  # noqa
+    demographic_fields,
+    demographic_list,
+    demographic_response
+)
+
+
+@demographic_api.route('/')
+class DemographicList(Resource):
+    @demographic_api.marshal_with(demographic_list)
+    def get(self):
+        """
+        Get all demographics
+
+        Returns paginated demographics
+        """
+        demographics = models.Demographic.query.all()
+        return kf_response(demographics, 200, '{} demographics'.
+                           format(len(demographics)))
+
+    @demographic_api.expect(demographic_fields, validate=True)
+    @demographic_api.marshal_with(demographic_response, code=201)
+    @demographic_api.doc(responses={201: 'demographic created',
+                                    400: 'demographic invalid input'})
+    def post(self):
+        """
+        Create a new demographic
+
+        Creates a new demographic and assigns a Kids First id
+        """
+
+        # Create the Demographic instance
+        body = request.json
+        demographic = models.Demographic(**body)
+
+        # Try saving to db
+        try:
+            db.session.add(demographic)
+            db.session.commit()
+        # Catch error and rollback db
+        except IntegrityError as e:
+            db.session.rollback()
+            context = {'method': 'create', 'entity': 'demographic',
+                       'ref_entity': 'participant', 'exception': e,
+                       'payload': body}
+            # raise BadInputError(context=context)
```

Comment:
Left this comment in here to show that error handling could also be handled with custom exceptions. With this approach, we would throw a custom exception and then handle it at the API level using the api.errorhandler decorator. Decided to avoid this approach for now due to some related known bugs with flask_restplus errorhandling. See  https://github.com/noirbizarre/flask-restplus/issues